server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -675,6 +681,10 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         // Remote Store
         actions.register(RestoreRemoteStoreAction.INSTANCE, TransportRestoreRemoteStoreAction.class);
 
+        // Decommission actions
+        actions.register(PutDecommissionAction.INSTANCE, TransportPutDecommissionAction.class);
```

Comment:
IMO, it can just be `DecommissionAction`

Reply:
Named it `PutDecommissionAction` to be in consistent with what is followed in repository for other features (eg. `PutRepositoryAction`, etc). Also, naming it  `PutDecommissionAction` clearly calls out that the action will create a mutating change

Reply:
Keeping it `DecommissionAction` for PUT request, `GetDecommissionStateAction` for GET request and `DeleteDecommissionStateAction` for DELETE request(PR for DELETE will be raised separately)
(idea is to not have two verbs clubbed together, in this case PutDecommission had two verbs)

@sachinpkale @shwetathareja @Bukhtawar lmk if this sounds ok?